pkg/agent/util/net_windows.go

```diff
@@ -579,3 +582,34 @@ func parseRoutes(routesStr string) ([]Route, error) {
 	}
 	return routes, nil
 }
+
+// GetVirtualInterfaceByName returns the virtual interface created by Windows when plugging the specified interface
+// on the HNSNetwork. If the virtual interface is not exist as well, the OVS bridge interface is returned. This function
+// is used for restart agent case.
+func GetVirtualInterfaceByName(ifaceName, ovsBridgeName string) (*net.Interface, error) {
+	vIfName := fmt.Sprintf("vEthernet (%s)", ifaceName)
+	link, err := net.InterfaceByName(vIfName)
+	if err != nil {
+		if !strings.Contains(err.Error(), "no such network interface") {
+			return nil, err
+		}
+		return net.InterfaceByName(ovsBridgeName)
+	}
+	return link, nil
+}
+
+func CreateNetNatOnHost(subnetCIDR *net.IPNet) error {
+	netNatName := "antrea-nat"
+	cmd := fmt.Sprintf(`Remove-NetNat -Name %s -Confirm:$false`, netNatName)
+	if _, err := ps.RunCommand(cmd); err != nil {
```

Comment:
Can we do "create if not found" instead of "remove and create" to be less disruptive?

Reply:
To be safe, it can compare whether the value is expected first, then decide whether to remove and create.

Reply:
Get-NetNat doesn't support using "InternalIPInterfaceAddressPrefix" as filter. I would try to compare with the result fields